pkg/controllers/deprovisioning/suite_test.go

```diff
@@ -720,11 +722,13 @@ var _ = Describe("Replace Nodes", func() {
 		Expect(env.Client.Get(ctx, client.ObjectKeyFromObject(node), node)).To(Succeed())
 
 		// consolidation won't delete the old node until the new node is ready
-		wg := ExpectMakeNewNodesReady(ctx, env.Client, 1, node)
+		var wg sync.WaitGroup
```

Comment:
This pattern is repeated over and over. Seems like a nontrivial burden for your average tester to understand this async verify wait thing. Wdyt about using a callback or something to encapsulate it?

Reply:
Toyed around with a different pattern by changing the existing code for a bit. Didn't come up with anything I was happy with. I'd rather take this in to fix the flakiness since it isn't really a degredation of the existing test code and then come back around to this later as a separate clean-up task.